synedrion/src/uint/signed.rs

```diff
@@ -532,82 +491,37 @@ where
     }
 }
 
-impl<T> Sub<&Signed<T>> for Signed<T>
+impl<'a, T> AddAssign<&'a Signed<T>> for Signed<T>
 where
-    T: crypto_bigint::Bounded + ConditionallySelectable + Encoding + Integer,
+    T: ConditionallySelectable + crypto_bigint::Bounded + Encoding + Integer,
 {
-    type Output = Self;
-    fn sub(self, rhs: &Self) -> Self::Output {
-        self.checked_add(&-rhs)
-            .expect("does not overflow by the construction of the arguments")
+    fn add_assign(&mut self, rhs: &'a Signed<T>) {
+        // TODO: implement properly
+        *self = self.checked_add(rhs).unwrap()
     }
 }
 
-impl<T> Mul<Signed<T>> for Signed<T>
+impl<'a, T> MulAssign<&'a Signed<T>> for Signed<T>
 where
-    T: Integer + Encoding + crypto_bigint::Bounded + ConditionallySelectable,
+    T: ConditionallySelectable + crypto_bigint::Bounded + Encoding + Integer,
 {
-    type Output = Self;
-    fn mul(self, rhs: Self) -> Self::Output {
-        self.checked_mul(&rhs)
-            .expect("does not overflow by the construction of the arguments")
+    fn mul_assign(&mut self, rhs: &'a Signed<T>) {
+        // TODO: implement properly
+        *self = self.checked_mul(rhs).unwrap()
     }
 }
 
-impl<T> Mul<&Signed<T>> for Signed<T>
+impl<T> Mul<Signed<T>> for Signed<T>
 where
     T: Integer + Encoding + crypto_bigint::Bounded + ConditionallySelectable,
 {
     type Output = Self;
-    fn mul(self, rhs: &Self) -> Self::Output {
-        self.checked_mul(rhs)
+    fn mul(self, rhs: Self) -> Self::Output {
+        self.checked_mul(&rhs)
             .expect("does not overflow by the construction of the arguments")
     }
 }
 
-impl<T> core::iter::Sum for Signed<T>
-where
-    T: Integer + Encoding + crypto_bigint::Bounded,
-{
-    fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
-        iter.reduce(|x, y| x.checked_add(&y).unwrap())
-            .unwrap_or(Self::default())
-    }
-}
-
-impl<'a, T> core::iter::Sum<&'a Self> for Signed<T>
-where
-    T: Integer + Encoding + crypto_bigint::Bounded,
-{
-    fn sum<I: Iterator<Item = &'a Self>>(iter: I) -> Self {
-        iter.cloned().sum()
-    }
-}
-
-impl<T> PartialOrd for Signed<T>
```

Comment:
Not needed or…?

Reply:
Since the code compiles, and `Signed` is `pub (crate)`, seems that yes, it's not needed.